### PR TITLE
Fix various health-server probing bugs.

### PR DESCRIPTION
**Pull request by marseel**
Fixes https://github.com/cilium/cilium/pull/29566 - The first issue mentioned in commit message was triggered by this change, all other issues existed before.

Commit message for convenience:

```
There were three issues with health-reporting/probing:
- Whenever node was updated, it was received in nodesAdded
and was overriding icmp result reporting node as unreachable
- If Icmp probe stopped working and there were no node updates,
it was reporting node as healthy even though probe was failing.
- Http prober was not triggered at the start and only after
  probeInterval.
```

## How to test it

- Triggering node update resets ICMP result: `kubectl label node kind-worker somelabel=label`
- ICMP results are not updated after first successful probe: 
```
docker exec -it KIND_WORKER_CONTAINER_ID /bin/sh
echo "1" >  /proc/sys/net/ipv4/icmp_echo_ignore_all
```

```release-note
Fixed health probing where ICMP probe was incorrectly reporting node as unreachable or reporting unreachable node as reachable in some cases.
```

### CODE REVIEW

File: pkg/health/server/prober.go [hunk 1]
```diff
@@ -216,11 +214,31 @@ func (p *prober) setNodes(added nodeMap, removed nodeMap) {
...
+
+	for _, status := range p.results {
+		if status.Icmp == nil {
+			status.Icmp = &models.ConnectivityStatus{}
+			status.Icmp.Status = "Connection timed out"
+		}
+	}
+}
+
+func (p *prober) cleanIcmpStatus() {
```
Comment (christarazi) [inline pkg/health/server/prober.go]: Nit: "Clear" is more accurate here, but this is a very nitty nit.

**Review by christarazi — APPROVED** [review]
Thanks for the fix. Could you reference the "broken" commit in the commit msg?

File: pkg/health/server/prober.go (hunk 1 above)
Comment (marseel) [reply pkg/health/server/prober.go]: done and done

**Comment by nbusseneau** [comment]
This PR was marked for backport to 1.12 but depends on changes made in #29745 which were not backported to 1.12. Marking as `backport/author` to let you decide what to do :)

**Comment by nbusseneau** [comment]
Synced with @marseel offline: we are not going to backport to 1.12 / 1.13 / 1.14 in the end.

**Pull request merged**